spec/registry.md

```diff
@@ -530,34 +530,24 @@ for examples.
 
 #### Determining Exact Literal Match
 
-> [!IMPORTANT]
-> The exact behavior of exact literal match is currently only well defined for non-zero-filled
-> integer values.
-> Functions that use fraction digits or significant digits might work in specific
-> implementation-defined ways.
-> Users should avoid depending on these types of keys in message selection in this release.
-
+If the numeric value of `resolvedSelector` is a non-negative integer
+and none of the following options are set for `resolvedSelector`,
+the serialized form of the numeric value MUST match the ABNF for `digit-size-option`,
+representing its decimal value:
+- `minimumFractionDigits`
+- `minimumIntegerDigits`
+- `minimumSignificantDigits`
+- `notation`
+- `style`
```

Comment:
I think this narrows the range we accept (to omit negative numbers), which isn't strictly necessary. We can repeat the relevant parts of the ABNF as needed.

This also forgets to mention when the `resolvedSelector` matches the `key`, which is kind of important. 

Finally, I would guess that `maximumXXXDigits` options would also be ignored?

Reply:
I'm ambivalent about including negative numbers. One ambiguity that opens up is whether `-0` and `0` are distinct.

The `key` comparison is handled in Number Selection.

You're right about missing `maximumSignificantDigits`; added that along with `fractionDigits` for `:currency`. There is no `maximumIntegerDigits`, and `maximumFractionDigits` isn't relevant when the value is an integer.